CI/LEGACY/travis/doxygen.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+TOP_DIR=$(pwd)
+
+set -x
+
+echo_red() { printf "\033[1;31m$*\033[m\n"; }
+echo_green() { printf "\033[1;32m$*\033[m\n"; }
+
+############################################################################
+# Check if the documentation will be generated w/o warnings or errors
+############################################################################
+pushd ${TOP_DIR}/build
+(cd doc && ! make doc 2>&1 | grep -E "warning|error") || {
+        echo_red "Documentation incomplete or errors in the generation of it have occured!"
+        exit 1
+}
+
+echo_green "Documentation was generated successfully!"
+
+############################################################################
+# If the current build is not a pull request and it is on master the 
+# documentation will be pushed to the gh-pages branch
+############################################################################
+if [[ "${IS_PULL_REQUEST}" == "False" && "${BRANCH_NAME}" == "master" ]]
```

Comment:
Could we add an extra check here to also verify if the branch is "main"? (After merging all the PRs we will change the default branch to 'main')